src/codeSigning/upload.ts

```diff
@@ -4,23 +4,18 @@
  * Licensed under the BSD 3-Clause license.
  * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
  */
-
-import AWS from 'aws-sdk';
+import { PutObjectCommand, PutObjectCommandOutput, S3 } from '@aws-sdk/client-s3';
+import { NodeHttpHandler, NodeHttpHandlerOptions } from '@smithy/node-http-handler';
 import { Agents } from 'got';
-import { WebIdentityCredentials } from 'aws-sdk';
 import { api } from './packAndSign.js';
-import ClientConfiguration = WebIdentityCredentials.ClientConfiguration;
 
-export async function putObject(bucket: string, key: string, body: string): Promise<AWS.S3.PutObjectOutput> {
-  return new Promise((resolve, reject) => {
-    const agent = api.getAgentForUri('https://s3.amazonaws.com') as Agents;
-    const s3 = new AWS.S3({
-      httpOptions: { agent: agent.http },
-      httpsOptions: { agent: agent.https },
-    } as ClientConfiguration);
-    s3.putObject({ Bucket: bucket, Key: key, Body: body }, (err, resp) => {
-      if (err) reject(err);
-      if (resp) resolve(resp);
-    });
+export async function putObject(bucket: string, key: string, body: string): Promise<PutObjectCommandOutput> {
+  const agent = api.getAgentForUri('https://s3.amazonaws.com') as Agents;
+  const s3 = new S3({
+    requestHandler: new NodeHttpHandler({
+      httpAgent: agent.http,
+      httpsAgent: agent.https,
+    } as NodeHttpHandlerOptions),
   });
+  return s3.send(new PutObjectCommand({ Bucket: bucket, Key: key, Body: body }));
```

Comment:
This appears to be the preferred way to call commands now, even though the old style does technically work, my experience was that I had to change it to the new style in order for the `upload.test.ts` mocking to work properly. If we want to stick to the old style, I can do more testing to see if maybe I was just being foolish and it works properly after all.